### PR TITLE
Support for `torch.float` weighted networks for FID and KID calculations.

### PR DESCRIPTION
**Pull request by furkan-celik**
## What does this PR do?
For evaluation of generated images in terms of how similar looking they are to the real images metrics FID is heavily used. However as pointed out in several papers, FID has some inherent flaws one of which is ImageNet InceptionV3 model is not suitable for all domains. Domains like medical MRI might be different enough to ImageNet that activation functions are not relevant to make FID comparison. For cases like this different domains use different variants of InceptionV3 model as feature extractor.

However, current implementation of FID and KID in torchmetrics only supports models with byteTensor weights as it is optimized for the tochmetrics' InceptionV3 model. In this PR I have given a bit more control to the developer so that it is possible to have better support for custom feature extractors with different dtypes and input sizes.

Details of this issue can be found in the following paper:
Liu, Shaohui, et al. "An improved evaluation framework for generative adversarial networks." arXiv preprint arXiv:1803.07474 (2018).

Fixes #\<issue_number>

<details>
  <summary>Before submitting</summary>

- [X I had issues with my implementations and didn't found a solution] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [+] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [+] Did you make sure to **update the docs**?
- [+] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2483.org.readthedocs.build/en/2483/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: src/torchmetrics/image/fid.py [hunk 1]
```diff
@@ -198,6 +198,15 @@ class FrechetInceptionDistance(Metric):
     flag ``real`` determines if the images should update the statistics of the real distribution or the
     fake distribution.
 
+    Using custom feature extractor is also possible. One can give a torch.nn.Module as `feature` argument. This
+    custom feature extractor is expected to have output shape of ``(1, num_features)`` This would change the
+    used feature extractor from default (Inception v3) to the given network. In case network doesn't have
+    ``num_features`` attribute, a random tensor will be given to the network to infer feature dimensionality.
+    Size of this tensor can be controlled by `input_img_size` argument and type of the tensor can be controlled
```
[the remark this answers is not in the document]
Comment (furkan-celik) [reply src/torchmetrics/image/fid.py]: fixed this

[the remark this answers is not in the document]
Comment (Borda) [reply src/torchmetrics/image/fid.py]: btw you can also just accept suggestion to make your life easier :)

File: src/torchmetrics/image/fid.py [hunk 2]
```diff
@@ -284,9 +295,16 @@ def __init__(
         feature: Union[int, Module] = 2048,
         reset_real_features: bool = True,
         normalize: bool = False,
+        input_img_size: Tuple[int, int, int, int] = (1, 3, 299, 299),
```
Comment (SkafteNicki) [inline src/torchmetrics/image/fid.py:298]: we just need the user to provide the image dimension, the 1 batch dimension we can add internally
```suggestion
        input_img_size: Tuple[int, int, int] = (3, 299, 299),
```

File: src/torchmetrics/image/fid.py [hunk 3]
```diff
@@ -304,10 +322,14 @@ def __init__(
 
         elif isinstance(feature, Module):
             self.inception = feature
+            self.used_custom_model = True
             if hasattr(self.inception, "num_features"):
                 num_features = self.inception.num_features
             else:
-                dummy_image = torch.randint(0, 255, (1, 3, 299, 299), dtype=torch.uint8)
+                if self.normalize:
+                    dummy_image = torch.rand(input_img_size, dtype=torch.float32)
```
Comment (SkafteNicki) [inline src/torchmetrics/image/fid.py:330]:
```suggestion
                    dummy_image = torch.rand(1, *input_img_size, dtype=torch.float32)
```

File: src/torchmetrics/image/fid.py [hunk 4]
```diff
@@ -304,10 +322,14 @@ def __init__(
...
+                else:
+                    dummy_image = torch.randint(0, 255, input_img_size, dtype=torch.uint8)
```
Comment (SkafteNicki) [inline src/torchmetrics/image/fid.py:332]:
```suggestion
                    dummy_image = torch.randint(0, 255, (1, *input_img_size), dtype=torch.uint8)
```

**Pull request merged**